src/app/detail/[task_id]/[user_type]/page.tsx

```diff
@@ -58,6 +69,18 @@ async function getSignedUrlUpload(token: string, fileName: string) {
   return data.signedUrl
 }
 
+async function getActivities(token: string, taskId: string): Promise<LogResponse[]> {
+  const res = await fetch(`${apiUrl}/api/tasks/${taskId}/activity-logs/?token=${token}`)
+  const data = await res.json()
+  return data.data
+}
+
+async function getTokenPayload(token: string): Promise<Token> {
+  const copilotClient = new CopilotAPI(token)
+  const payload = TokenSchema.parse(await copilotClient.getTokenPayload())
+  return payload
+}
```

Comment:
`CopilotAPI#getTokenPayload` itself returns a value parsed by TokenSchema so reparsing it doesn't seem necessary here. Also `null` value isn't well handled and would cause the app to crash.

![Screenshot from 2024-06-20 18-12-12](https://github.com/copilot-platforms/tasks-app/assets/59971845/a2f8b935-5145-4092-aba6-f96a69bbebea)